nasa/earthdata-search
# EDSC-2930: Adding update subscription functionality

# Overview

### What is the feature?

Adds the ability for a user to update their subscription.

### What is the Solution?

An action was added to handle the graphql request. The updated subscription is immediately updated in the store upon successful update. The delete subscription behavior was changed slightly to manually edit the store upon successful deletion.

### What areas of the application does this impact?

The focused collection subscription panel

# Testing

### Reproduction steps

- **Environment for testing:** Any
- **Collection to test with:** Any that support subscriptions

1. Add a subscription
2. Change granule search parameters
3. Click Update button on the subscription
4. Confirm the query for the subscription is updated
5. Click Delete on the subscription
6. Confirm the subscription was deleted correctly

# Checklist
